### ISSUE TITLE
Onion.Maui.GoogleMaps 6.0.1 iOS

### ISSUE THREAD

**Issue by mikebikerider** [themronion/Maui.GoogleMaps#73]
Build 6.0.1 adds new dependency for iOS - AdamE.Google.iOS.Maps 9.1.1.2. I updated my iOS project from 6.0.0 to 6.0.1. The AdamE.Google.iOS.Maps was not automatically installed by the VS2022 NuGet Manager. Building the project failed with errors pointing to the missing NuGet. I looked up the AdamE.Google.iOS.Maps 9.1.1.2 - 156 downloads, tiny compared to 70635 downloads for Onion.Maui.GoogleMaps. The AdamE.Goolge.iOS.Maps also looks like a work in progress and requiring hacks.  Instead of installing the missing NuGet I downgraded back to 6.0.0. 
What is the reason for adding a new dependency for iOS?

**Comment by themronion** [comment]
> VS2022

I am shocked that people still use this garbage

> tiny compared to 70635 downloads for Onion.Maui.GoogleMaps

You are comparing wrong things

> What is the reason for adding a new dependency for iOS?

Well, if you are comfortable using a 3 year old iOS sdk then cheers, mate 🥂. More info here: #72

**Comment by mikebikerider** [comment]
> I am shocked that people still use this garbage

What should people use instead?

I manually added missing dependency NuGets and unistalled Xamarin.Google.iOS.Maps.  Built the project - 576 errors most pointing to the AdamE.Google.iOS.Maps library. It looks like just importing this NuGet into VS2022 Pro build 17.11.5 (garbage?) is not enough. Some tweaks required.

Downgraded again:
In Package Manager uninstalled AdamE.Google.iOS.Maps. If left the build will fail. 
Selected Onion.Maui.GoogleMaps  6.0.0, clicked Install. VS2022 prompted to install the deprecated Xamarin.Google.iOS.Maps 6.0.1.1. Rebuilt the project. The output: 1 succeeded, 0 failed.

When I was upgrading Onion.Maui.GoogleMaps to build 6.0.1 the day before, I didn't actively seek this update. There were 3 other updates Microsoft.Maui.Controls and Microsoft.Maui.Controls.Compatibility among them. I saw Onion.Maui.GoogleMaps update available and selected it too. I had no idea it is more than a minor update let alone it would not compile. 

The number of Adam.Google.iOS.Maps downloads jumped from 165 to 382. Hopefully somebody here will report success and share how to make it work.

**Comment by themronion** [comment]
Nah, man, there is something wrong in your project. You shouldn’t manually install/uninstall transitive dependencies. And also clearing bin/obj folders is always a good helper

**Comment by mikebikerider** [comment]
I did clear bin/obj folders each time I tried to update. I was also clearing the hidden .vs folder.
The project is a published production application. It is a big project. The errors I was getting were about missing resources in AppData\Local\XamarinBuildDownloadCache\GMps-9.1.1.

**Comment by mikebikerider** [comment]
I tried following:
In VS2022 Pro version 17.11.5 running on a Windows 11 machine I normally don't use for MAUI development I created a new MAUI project iOS only. I kept the MainPage with 'Hello, World!' and 'Welcome to .Net Multi-platform App UI' labels and 'Click me' button.
I use Mac M2 mini macOS Sonoma 14.6.1, Xcode 16.0 to build iOS binaries.
I can import Onion.Maui.GoogleMaps 6.0.0 and the app compiles without errors.

Then I uninstalled Onion.Maui.GoogleMaps 6.0, cleared obj and bin folders, cleared Users\..\AppData\Local\ XamarinBuildDownloadCache\.
Imported Onion.Maui.GoogleMaps 6.0.1. The import looked normal, there was a prompt to install dependencies. I try to compile the project. VS2022 build runs forever or until I cancel it.
When I examine the new content in XamarinBuildDownloadCache it is different compared to 6.0.0. import. XamarinBuildDownloadCache contains a compressed folder GMps-9.1.1.tgz. I can examine the content of the compressed folder in Windows 11 File Explorer. VS2022 seems having problems with that, or wrong content.

The bottom line is I don't see it working.

Michael Rubinstein

**Comment by mikebikerider** [comment]
The issue is caused by the AdamE.Google.iOS.Maps 9.1.1.2 NuGet package. I found an identical issue posted at https://github.com/AdamEssenmacher/GoogleApisForiOSComponents/issues/21 . The two last posts there have a solution.

If not already installed on the development PC or Mac with Parallels VM,  download and Install 7-Zip freeware from 7-Zip<https://www.7-zip.org/>  .
Add the following section to the project file:

                <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('windows'))">
                                <XamarinBuildDownloadUser7ZipPath>C:\Program Files\7-Zip\7z.exe</XamarinBuildDownloadUser7ZipPath>
                </PropertyGroup>

I already had 7-Zip installed on my Windows 11 development machine. After I added the above PropertyGroup section to the project file the project with 6.0.1 Onion.Maui.GoogleMaps builds without errors and installs/runs on a physical device and on a simulator.

Thank you for the 6.0.1 update. I like the fresh look of Google Maps for iOS.

Michael Rubinstein (mikebikerider)
